pkg/controller/tenant/tenant_controller.go

```diff
@@ -180,24 +181,84 @@ func (tc *tenantController) syncTenant(tenantName string) (err error) {
 		klog.V(4).Infof("Finished initializing tenant %q (%v)", tenantName, time.Since(startTime))
 	}()
 
-	createFailures := []error{}
+	if err, done := tc.createNamespaces(tenantName); !done {
+		return err
+	}
 
-	// Create Default namespaces
-	for _, nsName := range tenant_default_namespaces {
+	if err, done := tc.createInitialRoleAndBinding(tenantName); !done {
+		return err
+	}
+
+	return nil
+}
 
+func (tc *tenantController) createNamespaces(tenantName string) (error, bool) {
+	failures := []error{}
+	for _, nsName := range tenantDefaultNamespaces {
 		ns := v1.Namespace{
 			ObjectMeta: metav1.ObjectMeta{Tenant: tenantName, Name: nsName},
 		}
 		if _, err := tc.kubeClient.CoreV1().NamespacesWithMultiTenancy(tenantName).Create(&ns); err != nil && !errors.IsAlreadyExists(err) {
-			createFailures = append(createFailures, err)
+			failures = append(failures, err)
 		}
 	}
-
-	if len(createFailures) != 0 {
-		ret := utilerrors.Flatten(utilerrors.NewAggregate(createFailures))
+	if len(failures) != 0 {
+		ret := utilerrors.Flatten(utilerrors.NewAggregate(failures))
 		klog.Errorf("Errors happened in tenant initialization of %v: %v", tenantName, ret)
-		return ret
+		return ret, false
 	}
+	return nil, true
+}
 
-	return nil
+func (tc *tenantController) createInitialRoleAndBinding(tenant string) (error, bool) {
+	// tenant admin act as cluster admin for tha tenant
+	roleName := initialClusterRoleName(tenant)
+	role := &rbacv1.ClusterRole{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:   roleName,
+			Tenant: tenant,
+		},
+		Rules: []rbacv1.PolicyRule{initialClusterRoleRules()},
+	}
+
+	binding := &rbacv1.ClusterRoleBinding{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:   initialClusterRoleBindingName(tenant),
+			Tenant: tenant,
+		},
+		Subjects: []rbacv1.Subject{
+			{Kind: rbacv1.UserKind, Name: "admin"},
+		},
+		RoleRef: rbacv1.RoleRef{APIGroup: rbacv1.GroupName, Kind: "ClusterRole", Name: initialClusterRoleName(tenant)},
+	}
+
+	var failures []error
+	if _, err := tc.kubeClient.RbacV1().ClusterRolesWithMultiTenancy(tenant).Create(role); err != nil && !errors.IsAlreadyExists(err) {
```

Comment:
Is there any problem if we proceed to create roleBinding without role being created?  I guess it should be fine, just want to double check.

Reply:
rolebinding can be created before or without the referred role. manually tested this.

Reply:
Great.